data/src/main/java/com/microsoft/azure/kusto/data/ObjectPopulator.java

```diff
@@ -0,0 +1,73 @@
+package com.microsoft.azure.kusto.data;
+
+import java.util.function.BiConsumer;
+
+/**
+ * Represents a single Kusto result column and its mapping to a java object. Used to extract the column from a {@link KustoResultSetTable} and set the
+ * value in a pojo. The column is defined by either its name or its ordinal (or both, in which case the ordinal is used).
+ * 
+ * @param <R>
+ *            the pojo this column needs to set its value to
+ * 			
+ * @param <C>
+ *            the Java class returned by the ADX column type
+ * @param <T>
+ *            type of ADX column
+ */
+class ObjectPopulator<R, C, T extends KustoType<C>> {
```

Comment:
I believe you already changed the name of this class, but doesn't it represent a KustoResultColumnPopulator? If so, I'd think that name makes most sense.

Reply:
ok.